pkg/cleanup/types/built_in.go

```diff
@@ -114,11 +116,10 @@ func scServiceBindingCleanup(spec, actual *unstructured.Unstructured) (runtime.O
 		return nil, err
 	}
 
-	sbSpec.Spec.ExternalID = sbActual.Spec.ExternalID
-	if sbActual.Spec.UserInfo != nil {
-		sbSpec.Spec.UserInfo = sbActual.Spec.UserInfo.DeepCopy()
```

Comment:
It's DeepCopyed before it gets here (in speccheck.go), and the ServiceInstance cleanup wasn't doing this (see below).

PS Maybe cleanup wrong name?